File: benchmarks/comprehensive/python/arithmetic_heavy.py
import time

ITERATIONS = 50000

def mandelbrot(cx, cy, max_iter):
    x = 0.0
    y = 0.0
    i = 0
    
    while i < max_iter:
        x2 = x * x
        y2 = y * y
        
        if x2 + y2 > 4.0:
            return i
        
        xtemp = x2 - y2 + cx
        y = 2.0 * x * y + cy
        x = xtemp
        i = i + 1
    
    return max_iter

start_time = time.time()
sum_val = 0

# Sample points in mandelbrot set
j = 0
while j < ITERATIONS:
    x = (j % 100) * 0.01 - 0.5
    y = (j // 100) * 0.01 - 0.5
    sum_val = sum_val + mandelbrot(x, y, 20)
    j = j + 1

end_time = time.time()

print(f"Arithmetic heavy result: {sum_val}")
print(f"Time: {(end_time - start_time) * 1000:.2f}ms")

File: benchmarks/comprehensive/javascript/arithmetic_heavy.js
const ITERATIONS = 50000;

function mandelbrot(cx, cy, maxIter) {
    let x = 0.0;
    let y = 0.0;
    let i = 0;
    
    while (i < maxIter) {
        const x2 = x * x;
        const y2 = y * y;
        
        if (x2 + y2 > 4.0) {
            return i;
        }
        
        const xtemp = x2 - y2 + cx;
        y = 2.0 * x * y + cy;
        x = xtemp;
        i = i + 1;
    }
    
    return maxIter;
}

const startTime = Date.now();
let sum = 0;

// Sample points in mandelbrot set
let j = 0;
while (j < ITERATIONS) {
    const x = (j % 100) * 0.01 - 0.5;
    const y = Math.floor(j / 100) * 0.01 - 0.5;
    sum = sum + mandelbrot(x, y, 20);
    j = j + 1;
}

const endTime = Date.now();

console.log(`Arithmetic heavy result: ${sum}`);
console.log(`Time: ${endTime - startTime}ms`);